src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -191,9 +191,7 @@ public function set($id, $service, $scope = self::SCOPE_CONTAINER)
             $this->scopedServices[$scope][$id] = $service;
         }
 
-        if (isset($this->aliases[$id])) {
-            unset($this->aliases[$id]);
-        }
+        unset($this->aliases[$id], $this->methodMap[$id]);
```

Comment:
For the record, I'd be :-1: now, because that would trigger a not-required copy-on-write for `$this->methodMap`.